File: frontend/demo/component/combobox/combo-box-readonly-and-disabled.ts
import 'Frontend/demo/init'; // hidden-source-line
import '@vaadin/combo-box';
import '@vaadin/horizontal-layout';
import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import { applyTheme } from 'Frontend/demo/theme';

@customElement('combo-box-readonly-and-disabled')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    applyTheme(root);
    return root;
  }

  protected override render() {
    return html`
      <vaadin-horizontal-layout theme="spacing">
        <!-- tag::snippet[] -->
        <vaadin-combo-box readonly label="Read-only" .items="${['Value']}" value="Value">
        </vaadin-combo-box>

        <vaadin-combo-box disabled label="Disabled"></vaadin-combo-box>
        <!-- end::snippet[] -->
      </vaadin-horizontal-layout>
    `;
  }
}

File: src/main/java/com/vaadin/demo/component/combobox/ComboBoxReadonlyAndDisabled.java
package com.vaadin.demo.component.combobox;

import com.vaadin.flow.component.combobox.ComboBox;
import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
import com.vaadin.flow.router.Route;
import com.vaadin.demo.DemoExporter; // hidden-source-line

@Route("combo-box-readonly-and-disabled")
public class ComboBoxReadonlyAndDisabled extends HorizontalLayout {

    public ComboBoxReadonlyAndDisabled() {
        setPadding(false);

        // tag::snippet[]
        ComboBox<String> readonlyField = new ComboBox<>();
        readonlyField.setReadOnly(true);
        readonlyField.setLabel("Read-only");
        readonlyField.setItems("Value");
        readonlyField.setValue("Value");

        ComboBox<String> disabledField = new ComboBox<>();
        disabledField.setEnabled(false);
        disabledField.setLabel("Disabled");
        // end::snippet[]
        add(readonlyField, disabledField);
    }

    public static class Exporter extends DemoExporter<ComboBoxReadonlyAndDisabled> { // hidden-source-line
    } // hidden-source-line
}
